tests/test_typing.py

```diff
@@ -25,7 +25,20 @@ class UserFactory(factory.Factory[User]):
             class Meta:
                 model = User
 
-        result: User
-        result = UserFactory.build()
-        result = UserFactory.create()
-        self.assertEqual(result.name, "John Doe")
+        assert_type(UserFactory.build(), User)
+        assert_type(UserFactory.create(), User)
+        assert_type(UserFactory.build_batch(2), list[User])
+        assert_type(UserFactory.create_batch(2), list[User])
+        self.assertEqual(UserFactory.create().name, "John Doe")
+
+    def test_dict_factory(self) -> None:
+        class Pet(factory.DictFactory):
+            species = "dog"
+            name = "rover"
+
+        self.assertTrue(isinstance(Pet.build(), dict))
+        self.assertTrue(isinstance(Pet.create(), dict))
+
+    def test_list_factory(self) -> None:
+        self.assertTrue(isinstance(factory.ListFactory.build(), list))
+        self.assertTrue(isinstance(factory.ListFactory.create(), list))
```

Comment:
These can use the `self.assertIsInstance` method from unittest.
```suggestion
        self.assertIsInstance(Pet.build(), dict)
        self.assertIsInstance(Pet.create(), dict)

    def test_list_factory(self) -> None:
        self.assertIsInstance(factory.ListFactory.build(), list)
        self.assertIsInstance(factory.ListFactory.create(), list)
```